collect_app/src/main/java/org/odk/collect/android/widgets/utilities/GeoPolyDialogFragment.kt

```diff
@@ -20,18 +21,22 @@ class GeoPolyDialogFragment(viewModelFactory: ViewModelProvider.Factory) :
             GeoPolyFragment.REQUEST_GEOPOLY,
             this
         ) { _, result ->
-            val result = result.getString(GeoPolyFragment.RESULT_GEOPOLY)
-            if (result != null) {
-                onAnswer(StringData(result))
+            val geopolyChange = result.getString(GeoPolyFragment.RESULT_GEOPOLY_CHANGE)
+            val geopoly = result.getString(GeoPolyFragment.RESULT_GEOPOLY)
+            val incremental = FormEntryPromptUtils.getBindAttribute(prompt, "incremental")
+
+            if (incremental == "true" && geopolyChange != null) {
+                onAnswer(StringData(geopolyChange), dismiss = false, validate = true)
+            } else if (geopoly != null) {
+                onAnswer(StringData(geopoly))
             } else {
                 dismiss()
```

Comment:
If it's a new point (`result` = `geotrace_change` but `incremental` is `false`/`null` then it will end up here closing the map biew. SO basicly after adding the first point it will be closed.